balancer/weightedroundrobin/weightedroundrobin.go

```diff
@@ -19,11 +19,47 @@
 // Package weightedroundrobin defines a weighted roundrobin balancer.
 package weightedroundrobin
 
+import (
+	"google.golang.org/grpc/resolver"
+)
+
 // Name is the name of weighted_round_robin balancer.
 const Name = "weighted_round_robin"
 
-// AddrInfo will be stored inside Address metadata in order to use weighted roundrobin
-// balancer.
+// attributeKey is the type used as the key to store AddrInfo in the Attributes
+// field of resolver.Address.
+type attributeKey struct{}
+
+// AddrInfo will be stored inside Address metadata in order to use weighted
+// roundrobin balancer.
 type AddrInfo struct {
 	Weight uint32
 }
+
+// SetAddrInfo returns a copy of addr in which the Attributes field is updated
+// with addrInfo.
+//
+// This is an EXPERIMENTAL API.
+func SetAddrInfo(addrInfo AddrInfo, addr resolver.Address) resolver.Address {
+	newAddr := addr
+	newAddr.Attributes = addr.Attributes.WithValues(attributeKey{}, addrInfo)
+	return newAddr
+}
+
+// GetAddrInfo returns the AddrInfo stored in the Attributes fields of addr.
+// Returns nil if no AddrInfo is present.
+//
+// This is an EXPERIMENTAL API.
+func GetAddrInfo(addr resolver.Address) AddrInfo {
+	if addr.Attributes == nil {
+		return AddrInfo{}
+	}
+	ai := addr.Attributes.Value(attributeKey{})
+	if ai == nil {
+		return AddrInfo{}
+	}
+	if _, ok := ai.(AddrInfo); !ok {
+		return AddrInfo{}
+	}
+	return ai.(AddrInfo)
```

Comment:
Since everything is nil-pointer-safe, I think this function can be simplified to something like:

```go
v := addr.Attributes.Value(attributesKey{})
ai, _ := v.(AddrInfo)  // ignore ok and use zero value AddrInfo if v is nil
return ai
```

Reply:
Done. Thanks.